### PR TITLE
github: added template for Pull Requests, issues and a CONTRIBUTION.md

### PR DESCRIPTION
**Pull request by toogley**
Followup PR to https://github.com/neomutt/neomutt/pull/312 - which i have accidentally closed.

@guyzmo @flatcap 

Maybe http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html is also worth to include into the PR tempalte?

### CODE REVIEW

**Comment by guyzmo** [comment]
don't forget to include the full text link to the contributing guide at the end of the templates, but I'm 👌 with how it looks.

**Comment by toogley** [comment]
@guyzmo Do you really think so? A notice to the CONTRIBUTING.md file is already generated by [Github](https://github.com/blog/1184-contributing-guidelines)

**Comment by flatcap** [comment]
Yes, I like tpope's commit message guidelines.

Yes, you're right, we don't need a link to CONTRIBUTING.md in the issue template because GitHub will create a link.  However, it doesn't seem to happen for pull-requests.

**Comment by toogley** [comment]
@neomutt/reviewers I have added some guidelines for good git commit messages. What do you think about them?

Especially this point is really important for me, because I found the readability **much** better if it was followed:

```
* your commits should be clear and concise. That means also you shouldn't sum up
  two features or two bug fixes into one commit. If you do, it makes both bug
  fixes or features harder to understand.
```

**Comment by guyzmo** [comment]
the reasoning beyond the atomic commit point is more importantly to help with git bisect. It's also why it's important to always merge commits that builds and can be tested. Then in case of a regression discovered later on, we don't end up testing a broken commit.

**Comment by guyzmo** [comment]
about commits, you should add to always add a line like:

```
Github-Issue: #<issue number>
```

or any keyword @flatcap has registered in the project's configuration, which once merged in the master branch, will automagically close the issue, and make the commit appear in the issue (as well as in the PR).

**Comment by flatcap** [comment]
GitHub docs (a good read :-)
- [Autolinked references and URLs](https://help.github.com/articles/autolinked-references-and-urls)
- [Closing issues via commit messages](https://help.github.com/articles/closing-issues-via-commit-messages/)

**Comment by d-k-c** [comment]
Seems pretty okay with me, a few nitpicks: don't capitalize words that shouldn't be like "Project" or "Bullets".

Also, usually commit titles use the active form, because they describe what the commit does/changes to the code base. For instance, your first commit "add templates for this, and this and that".

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+* What does this PR do?
+
+* Are there points in the code the reviewer needs to double check?
+
+* Why was this PR needed?
+
+* Screenshots (if relevant)
+
+* Does this PR meet the acceptance criteria?
+
```
Comment (guyzmo) [inline .github/PULL_REQUEST_TEMPLATE.md]: now that we've got (or are about to have) a style guide we could add:

- [ ] Code syntax follows the style guide

Comment (toogley) [inline .github/PULL_REQUEST_TEMPLATE.md]: fixed.

**Comment by toogley** [comment]
I have implemented all suggestions. What do you think?

**Comment by toogley** [comment]
@flatcap please don't merge it yet. i have forgotten that the first line of every commit message should be less than 50 chars so that github doesn't split the sentence in half.

**Comment by toogley** [comment]
@flatcap  fixed.

**Comment by flatcap** [comment]
Sorry, @toogley.  This was branched from [neomutt] and when I deleted that branch, this issue auto-closed itself.

**Comment by toogley** [comment]
@flatcap are there still issues with this PR?

**Pull request merged**